spec/services/storage/s3/uploader_spec.rb

```diff
@@ -1,84 +1,69 @@
 require 'rails_helper'
 
 RSpec.describe Storage::S3::Uploader do
-  let(:test_client) { Aws::S3::Client.new(stub_responses: stub_responses) }
-
-  before :each do
-    allow(Aws::S3::Client).to receive(:new).and_return(test_client)
-  end
-
+  let(:s3) { Aws::S3::Client.new(stub_responses: true) }
   let(:path) { file_fixture('lorem_ipsum.txt') }
   let(:key) { '28d/service-slug/upload-fingerprint' }
+  let(:downloader) { Storage::S3::Downloader.new(key: key) }
+  let(:subject) { described_class.new(path: path, key: key) }
 
-  subject do
-    described_class.new(path: path, key: key)
+  before :each do
+    allow(Aws::S3::Client).to receive(:new).and_return(s3)
   end
 
-  let(:downloader) { Storage::S3::Downloader.new(key: key) }
+  describe '#exists?' do
+    context 'when a file does not exist in s3' do
+      before do
+        s3.stub_responses(:head_object, 'NotFound')
+      end
 
-  describe '#upload' do
-    describe do
-      context do
-        let(:stub_responses) do
-          {
-            head_object: [ false, { content_length: 150 } ],
-            put_object: [{}],
-          }
-        end
+      it 'returns false' do
+        expect(subject.exists?).to eq(false)
+      end
+    end
 
-        it 'uploads file to s3' do
-          expect(subject.exists?).to be_falsey
-          subject.upload
-          expect(subject.exists?).to be_truthy
-        end
+    context 'when a file exists in s3' do
+      before do
+        s3.stub_responses(:head_object, {})
       end
 
-      context do
-        let(:stub_responses) do
-          {
-            head_object: [ { content_length: 150 } ],
-            put_object: [{}],
-            get_object: [{ body: "ce030d6aac29d4a5a8b03f7428ff4626" }]
-          }
-        end
+      it 'returns true' do
+        expect(subject.exists?).to eq(true)
+      end
+    end
+  end
 
-        it 'encrypts file contents' do
```

Comment:
I've lost this spec. Specifically testing that the file is encrypted. I can't see a good way of doing this without reverting to using a local storage solution instead of stubbing S3. We want to move away from the local storage.